Sources/InfomaniakCore/Extensions/EnsureFrozen.swift

```diff
@@ -0,0 +1,40 @@
+/*
+ Infomaniak Core - iOS
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import RealmSwift
+
+/// This property wrapper ensures that the passed object (or object collection) is frozen. In case the given object isn't frozen an assert is raised.
+@propertyWrapper
+public struct EnsureFrozen<Value> {
```

Comment:
Since we have access to Realm types, and thats not likely to change, we could specialize `Value`.

Something like
```swift
public struct EnsureFrozen<Value: ThreadConfined> {
```

So using it with some DTOs would fail at build time, a useful behaviour IMO.
It could also reduce the need to downcast in the debug code, something I have not checked.

Reply:
Great idea 👍